src/Domain/CampaignRenderCompatibilityChecker.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace MatchBot\Domain;
+
+use Assert\Assert;
+use Assert\LazyAssertionException;
+
+/**
+ * Checks that a campaign as rendered to an array by new matchbot code is compatible with how it was
+ * rendered by the SF API.
+ */
+class CampaignRenderCompatibilityChecker
```

Comment:
this is all very general code, its just checking if everything in one array matches another, so maybe could come from a library but not sure which library and writing from scratch was probably as quick as choosing a library.

Also it may get more complex soon as we might need to include allowances for some differences, e.g. for the campaign values that can change inside matchbot after the campaign is pulled from SF like number of donations.